cpp/include/cudf/strings/find.hpp

```diff
@@ -103,6 +104,28 @@ std::unique_ptr<column> contains(
   string_scalar const& target,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a column of boolean values for each string where true indicates
+ * the corresponding target string was found within that string in the provided column.
+ *
+ * The 'output[i] = true` if string `targets[i]` is found inside `strings[i]` otherwise
+ * `output[i] = false`.
+ * If `target[i]` is an empty string, true is returned for `output[i]`.
+ *
+ * Any null string entries return corresponding null entries in the output column.
+ *
+ * @throw cudf::logic_error if `strings.size() != targets.size()`.
+ *
+ * @param strings Strings instance for this operation.
+ * @param targets Strings column of targets to check row-wise in `strings`.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New type_id::BOOL8 column.
+ */
+std::unique_ptr<column> contains(
```

Comment:
Is there a reasoning behind calling this `contains`? C++20 associative containers and sets are getting a [`contains` method](https://en.cppreference.com/w/cpp/container/map/contains) and it might be confusing. Also, the Python API seems to call this `contains_multiple`? 

Not saying we should change it for sure ... just want to make sure that it has been thought through.

Reply:
I just re-read the description and I guess it makes more sense as it is an overload of the `cudf::strings::contains` API that just takes a scalar - still wondering about peoples thoughts on this though.

Reply:
Also, there is a non-regex version `cudf::strings::contains_re` to consider.
https://github.com/rapidsai/cudf/blob/4b4d96265110ff1c70e2ddff15c19e2b918e8c29/cpp/include/cudf/strings/contains.hpp#L50-L53
We can rename all of these but not sure if that sure be in this PR.
The `contains_multiple` was to be consistent with `startswith_multiple` and `endswith_multiple` which have the same pattern. It does make it easier to call the right API from Python/Cython by checking the parameter is either a scalar or a column.

Reply:
Yea, if there is precedence I would just say leave as is then. :+1: